File: PGS/Lv.1/3진법 뒤집기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <cmath>

using namespace std;

int solution(int n) {
	int answer = 0;
	string s = "";
	while (n) {
		s += n % 3 + '0';
		n /= 3;
	}
	reverse(s.begin(), s.end());

	for (int i = 0; i < s.size(); i++) {
		answer += (s[i] - '0') * pow(3, i);
	}

	return answer;
}

File: PGS/Lv.1/JAVA/3진법 뒤집기.java
class Solution {
    public int solution(int n) {
        
        String str = Integer.toString(n, 3);    // 1. 정수 -> 3진법 문자열
        String reversed = new StringBuilder(str).reverse().toString();  // 2. 3진법 문자열 뒤집기
        int answer = Integer.parseInt(reversed, 3);   // 3. 3진법 문자열 -> 정수
        
        return answer;
    }
}